File: 05_react/app-example/frontend/src/Product.js
import React from 'react';

class Product extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <div className="product">
                <img alt="product image" src={this.props.data.imageUrl}/>
                <div className="product-title">
                    {this.props.data.name}
                </div>
                <div className="product-description">{this.props.data.description}</div>
                <p>${this.props.data.price}</p>
            </div>
        );
    }
}

export default Product;

File: 05_react/app-example/src/main/java/com/anrymart/example/model/Product.java
package com.anrymart.example.model;

import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
public class Product {
    String name;
    String description;
    String imageUrl;
    double price;

    public Product(String name, String description, String imageUrl, double price) {
        this.name = name;
        this.description = description;
        this.imageUrl = imageUrl;
        this.price = price;
    }
}
